sdk/core/core-rest-pipeline/src/policies/retryPolicy.ts

```diff
@@ -0,0 +1,140 @@
+ */
+export function retryPolicy(
+  strategies: RetryStrategy[],
+  options: RetryPolicyOptions = { maxRetries: DEFAULT_MAX_RETRIES }
+): PipelinePolicy {
+  const logger = options.logger || retryPolicyLogger;
+  return {
+    name: retryPolicyName,
+    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      let response: PipelineResponse | undefined;
+      let responseError: RestError | undefined;
+      let retryCount = -1;
+
+      // eslint-disable-next-line no-constant-condition
+      retryRequest: while (true) {
+        retryCount += 1;
+        response = undefined;
+        responseError = undefined;
+
+        try {
+          logger.info(`Retry ${retryCount}: Attempting to send request`, request.requestId);
+          response = await next(request);
+          logger.info(`Retry ${retryCount}: Received a response from request`, request.requestId);
+        } catch (e) {
+          logger.error(`Retry ${retryCount}: Received an error from request`, request.requestId);
+          responseError = e as RestError;
+          if (!e || responseError.name !== "RestError") {
+            throw e;
+          }
+          response = responseError.response;
+        }
+
+        if (request.abortSignal?.aborted) {
+          logger.error(`Retry ${retryCount}: Request aborted.`);
+          const abortError = new AbortError();
+          throw abortError;
+        }
+
+        if (retryCount >= (options.maxRetries ?? DEFAULT_MAX_RETRIES)) {
+          logger.info(
+            `Retry ${retryCount}: Maximum retries reached. Returning the last received response, or throwing the last received error.`
+          );
+          if (response !== undefined) {
+            return response;
+          }
+          throw responseError;
+        }
+
+        logger.info(`Retry ${retryCount}: Processing ${strategies.length} retry strategies.`);
+
+        strategiesLoop: for (const strategy of strategies) {
+          const strategyLogger = strategy.logger || retryPolicyLogger;
+          strategyLogger.info(`Retry ${retryCount}: Processing retry strategy ${strategy.name}.`);
+
+          const modifiers = strategy.retry({
+            retryCount,
+            response,
+            responseError
+          });
+
+          if (modifiers.skipStrategy) {
+            strategyLogger.info(`Retry ${retryCount}: Skipped.`);
+            continue strategiesLoop;
+          }
+
+          const { errorToThrow, retryAfterInMs, redirectTo } = modifiers;
+
+          if (errorToThrow) {
+            strategyLogger.error(
+              `Retry ${retryCount}: Retry strategy ${strategy.name} throws error:`,
+              errorToThrow
+            );
+            throw errorToThrow;
+          }
+
+          if (retryAfterInMs) {
+            strategyLogger.info(
+              `Retry ${retryCount}: Retry strategy ${strategy.name} retries after ${retryAfterInMs}`
+            );
+            await delay(retryAfterInMs);
```

Comment:
Should we (or are we already) adding some randomness / jitter to the delay? 

https://stackoverflow.com/questions/46939285/why-is-random-jitter-applied-to-back-off-strategies

Reply:
Oh nevermind I think I see it here https://github.com/Azure/azure-sdk-for-js/pull/19078/files#diff-f2c983a6a6bb024ee11af07074179454d58d6b9284aa0c351caccf9a87e040b6R71 just missed it the first time

Reply:
Yeah was going to point you to here

https://github.com/Azure/azure-sdk-for-js/blob/b3801dde58aac593b1e904a4fa56f4b0f0d2d78f/sdk/core/core-rest-pipeline/src/retryStrategies/exponentialRetryStrategy.ts#L72

This is a good thing to ask about though, since the original version did not have jitter (and I can't remember if we fixed this in corev1)

Reply:
Jitter is part of the `exponentialRetryStrategy`, which is used by the `exponentialRetryPolicy`, and used in composition on the default retries and the system error retries.

Reply:
My concern would be that adding jitter to the retry policy itself may result in unexpected behaviors. Jitter can be easily added on the strategies needing jitter 🤔

Reply:
Yeah I don't think we should add jitter on top of what the strategy requests, so we're good with present behavior